runtime/runtime-params-estimator/test-contract/src/lib.rs

```diff
@@ -439,6 +442,187 @@ pub unsafe fn keccak512_10kib_10k() {
+    for _ in 0..1_000 {
+        alt_bn128_g1_sum(buffer.len() as u64, buffer.as_ptr() as *const u64 as u64, 0);
+    }
+}
+
+
+
+// Function to measure `alt_bn128_pairing_check_base` and `alt_bn128_pairing_check_element`. Also measures `base`, `write_register_base`,
+// and `write_register_byte`. However `g1_multiexp` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute pairing_check on 1 element 1k times.
+#[no_mangle]
+pub unsafe fn alt_bn128_pairing_check_1_1k() {
+    let buffer: [u8; 196] = [
+        1, 0, 0, 0, 80, 12, 4, 181, 61, 254, 153, 52, 127, 228, 174, 24, 144, 95, 235, 26, 197,
+        188, 219, 91, 4, 47, 98, 98, 202, 199, 94, 67, 211, 223, 197, 21, 65, 221, 184, 75, 69,
+        202, 13, 56, 6, 233, 217, 146, 159, 141, 116, 208, 81, 224, 146, 124, 150, 114, 218, 196,
+        192, 233, 253, 31, 130, 152, 144, 29, 34, 54, 229, 82, 80, 13, 200, 53, 254, 193, 250, 1,
+        205, 60, 38, 172, 237, 29, 18, 82, 187, 98, 113, 152, 184, 251, 223, 42, 104, 148, 253, 25,
+        79, 39, 165, 18, 195, 165, 215, 155, 168, 251, 250, 2, 215, 214, 193, 172, 187, 84, 54,
+        168, 27, 100, 161, 155, 144, 95, 199, 238, 88, 238, 202, 46, 247, 97, 33, 56, 78, 174, 171,
+        15, 245, 5, 121, 144, 88, 81, 102, 133, 118, 222, 81, 214, 74, 169, 27, 91, 27, 23, 80, 55,
+        43, 97, 101, 24, 168, 29, 75, 136, 229, 2, 55, 77, 60, 200, 227, 210, 172, 194, 232, 45,
+        151, 46, 248, 206, 193, 250, 145, 84, 78, 176, 74, 210, 0, 106, 168, 30,
+    ];
+    for _ in 0..1_000 {
+        alt_bn128_pairing_check(buffer.len() as u64, buffer.as_ptr() as *const u64 as u64);
+    }
+}
+// Function to measure `alt_bn128_g1_multiexp_base` and `alt_bn128_g1_multiexp_sublinear`. Also measures `base`, `write_register_base`,
+// and `write_register_byte`. However `g1_multiexp` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute pairing_check on 10 elements 1k times.
+#[no_mangle]
+pub unsafe fn alt_bn128_pairing_check_10_1k() {
+    let buffer: [u8; 1924] = [
+        10, 0, 0, 0, 223, 6, 200, 33, 166, 188, 103, 53, 200, 168, 35, 109, 73, 23, 33, 99, 238,
```

Comment:
How this buffer was produced? Does presence of large data buffer significantly affects other costs, such as compilation cost?

Reply:
This is a serialized vector of 10 (g1, g2) points. it should not affect other costs, because it is growing linear and pairing check is much more expensive (you could compare the one's cost with keccak256 for example).